### ISSUE TITLE
failed to acquire cache lock and failed to unlock

### ISSUE THREAD

**Issue by mosesyou** [Mashape/kong#2166]
### Summary
I use `kong:cassandra` this docker image,and use its default configuration.Then I add a POST method api and make 200 concurrent time post requests continuously.There are some error happend;
This api have 3000 tps when do not use kong, but only 200 now.

It seems that lua-resty-lock this module's problem, i turn up lua_shared_dict cache_locks 100k to 2m.
But still not work.
So,i dont know where the problem is.
Help please...
### Steps To Reproduce

1. run images kong:cassandra3&cassandra:2.2
2. add a api
3. use jmeter to post requests

error log:
```
database_cache.lua:146: get_or_set(): failed to acquire cache lock: timeout, client: x.x.x.x server: kong, request: "POST /producer/stockmsg HTTP/1.1", host: "x.x.x.x:port"
2017/03/07 03:49:01 [error] 112#0: *10925 [lua] database_cache.lua:165: get_or_set(): failed to unlock: unlocked, client: x.x.x.x, server: kong, request: "POST /producer/stockmsg HTTP/1.1", host: "x.x.x.x:port"
...
Constantly appear above the wrong
```

BTW:i use jmeter to post request.
### Additional Details & Logs

- Kong version (`0.9.0.0`)
- Operating System centos7&docker `1.10.3`
- kong docker images `kong:cassandra3`
- database `cassandra 2.2`

**Comment by mosesyou** [comment]
It seems that `lua-resty-lock` this module's problem, i turn up `lua_shared_dict cache_locks` 100k to 2m.
But still  not work.
So,i dont know where the problem is.
Help please...

**Comment by Tieske** [comment]
> - kong docker images `kong:cassandra3`
> - database `cassandra 2.2`

Which is it? Cassandra 3 or Cassandra 2.2?  Note that for Kong 0.9.x only Cassandra 2.x is supported. Kong 0.10 supports Cassandra 3.

Also Kong 0.9.x has seen quite some fixes, so latest for that version is 0.9.9. Have you tried that, or even better the (brand) new 0.10

**Comment by Tieske** [comment]
see #2186 and #2197 

closing this one
